framework/helpers/BaseArrayHelper.php

```diff
@@ -178,10 +178,15 @@ public static function merge($a, $b)
      * The possibility to pass an array of keys is available since version 2.0.4.
      * @param mixed $default the default value to be returned if the specified array key does not exist. Not used when
      * getting value from an object.
-     * @return mixed the value of the element if found, default value otherwise
+     * @return mixed the value of the element if found, default value otherwise.
+     * @throws InvalidParamException if $array is neither an array nor an object.
```

Comment:
this line existed in 2.0.10, so that is not new. https://github.com/yiisoft/yii2/blob/2.0.10/framework/helpers/BaseArrayHelper.php#L182